### PR TITLE
Convert `pp` commands to the rzshell

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Convert all `pp` commands to rzshell

**Test plan**

CI is green

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/1590

### CODE REVIEW

File: librz/core/cmd/cmd_print.c [hunk 1]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
 	return print_value_size(core, state, argc, argv, 8);
 }
+
+RZ_IPI RzCmdStatus rz_print_pattern0_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut8 *buf = (ut8 *)rz_debruijn_pattern(len, 0, NULL);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: check for null ptr

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+	for (int i = 0; i < len; i++) {
+		rz_cons_printf("%02x", 0);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: this outputs only zeros, why allocating?

File: librz/core/cmd/cmd_print.c [hunk 3]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+RZ_IPI RzCmdStatus rz_print_pattern2_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// TODO: honor cfg.bigendian
+	int min = (core->offset & 0xffff);
+	for (size_t i = 0; i < len; i++) {
+		rz_cons_printf("%04zx", i + min);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: maybe define `min` and `i` as `ut32` and remove the `%zx` and use only `%x`

File: librz/core/cmd/cmd_print.c [hunk 4]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+RZ_IPI RzCmdStatus rz_print_pattern4_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// TODO: honor cfg.bigendian
+	int min = (core->offset & UT32_MAX);
+	for (size_t i = 0; i < len; i++) {
+		rz_cons_printf("%08zx", i + min);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: same here.

File: librz/core/cmd/cmd_print.c [hunk 5]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+
+RZ_IPI RzCmdStatus rz_print_pattern1_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int min = (core->offset & 0xff);
+	for (size_t i = 0; i < len; i++) {
+		rz_cons_printf("%02zx", i + min);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: just use `ut8` for `min` and `i` make it `ut32`

File: librz/core/cmd/cmd_print.c [hunk 6]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern_latin_alphabet_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	size_t bs = 4;
+	ut8 *buf = calloc(bs, 1);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c:6504]: check for nullptr

File: librz/core/cmd/cmd_print.c [hunk 7]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern_debrujin_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut8 *buf = (ut8 *)rz_debruijn_pattern(len, 0, NULL);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c:6527]: check for nullptr

Comment (wargio) [reply librz/core/cmd/cmd_print.c:6527]: also why this is different from the one above?

File: librz/core/cmd/cmd_print.c [hunk 8]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+	for (int i = 0; i < len; i++) {
+		rz_cons_printf("%02x", 0xff);
+	}
+	rz_cons_newline();
+	free(buf);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: why allocating ?

File: librz/core/cmd/cmd_print.c [hunk 9]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+}
+
+RZ_IPI RzCmdStatus rz_print_pattern_num_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	size_t bs = 4;
+	ut8 *buf = calloc(bs, 1);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c:6560]: check for nullptr

File: librz/core/cmd/cmd_print.c [hunk 10]
```diff
@@ -6570,3 +6428,147 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+	for (int i = 0; i < len; i++) {
+		incDigitBuffer(buf, bs);
+		for (int j = 0; j < bs; j++) {
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c]: `i` and `j` needs to be `st64`

File: librz/core/cmd_descs/cmd_print.yaml [hunk 11]
```diff
@@ -461,6 +461,72 @@ commands:
...
+    subcommands:
+      - name: pp0
+        summary: Print buffer filled with zeroes
+        cname: print_pattern0
+        args:
+          - name: len
+            type: RZ_CMD_ARG_TYPE_NUM
+            optional: true
+      - name: pp1
+        summary: Print incremental byte pattern (honor lower bits of cufrent address and bsize)
```
Comment (wargio) [inline librz/core/cmd_descs/cmd_print.yaml]: this is not really clear, maybe explicitly say `block size`

**Review by wargio — CHANGES_REQUESTED** [review]

File: librz/core/cmd/cmd_print.c [hunk 12]
```diff
@@ -6570,3 +6428,155 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+
+RZ_IPI RzCmdStatus rz_print_pattern1_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	ut8 min = (core->offset & 0xff);
+	for (ut8 i = 0; i < len; i++) {
+		rz_cons_printf("%02x", i + min);
```
Comment (wargio) [inline librz/core/cmd/cmd_print.c:6446]: This loops indefinitely since i overflows

**Review by wargio — APPROVED** [review]
ok, besides the overflows. maybe limit the max len

File: librz/core/cmd/cmd_print.c [hunk 13]
```diff
@@ -6570,3 +6428,155 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
 }
+
+RZ_IPI RzCmdStatus rz_print_pattern0_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	for (int i = 0; i < len; i++) {
+		rz_cons_printf("%02x", 0);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: Just print 00 without the printf

File: librz/core/cmd/cmd_print.c [hunk 14]
```diff
@@ -6570,3 +6428,155 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// TODO: honor cfg.bigendian
+	ut64 min = (core->offset);
+	for (ut64 i = 0; i < len; i++) {
+		rz_cons_printf("%016" PFMT64x, i + min);
+	}
+	rz_cons_newline();
+	return RZ_CMD_STATUS_OK;
+}
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c:6495]: I suggest to refactor all these function into one that just accept a size argument. They are all essentially the same

File: librz/core/cmd/cmd_print.c [hunk 15]
```diff
@@ -6570,3 +6428,155 @@ RZ_IPI RzCmdStatus rz_print_value4_handler(RzCore *core, int argc, const char **
...
+
+RZ_IPI RzCmdStatus rz_print_pattern_oxff_handler(RzCore *core, int argc, const char **argv) {
+	st64 len = argc > 1 ? rz_num_math(core->num, argv[1]) : core->blocksize;
+	if (len < 1) {
+		RZ_LOG_ERROR("Invalid pattern length\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	for (int i = 0; i < len; i++) {
+		rz_cons_printf("%02x", 0xff);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: Just print ff ?

File: librz/core/cmd/cmd_print.c (hunk 14 above)
Comment (XVilka) [reply librz/core/cmd/cmd_print.c:6495]: They aren't the same due to the size and format string used. Of course, they can be generalized but that would introduce unnecessary complexity, IMHO. Lets keep this as is.

**Review by wargio — APPROVED** [review]
i'm not sure if there are tests for this, but i'm ok with the changes

**Pull request merged**